frontend/animal/VocabBrowser/App.js

```diff
@@ -16,7 +17,13 @@ class App extends Component {
             query: "",
         };
         this.updateQuery = _.debounce(() => props.store.updateQuery(this.state.query), 100);
+
+        this.tables = {
```

Comment:
To reduce duplicate code, I added a more generic VocabBrowser folder that imports the correct term table based on the vocabulary invoked. To render the correct table, I had to explicitly define all possible options: I don't love this as a setup, but couldn't figure out how to avoid this, and am open to any changes.

Reply:
I also just implemented a version of this which uses dynamic loading (to import the table I need instead of every table). To update a vocab in this version, you would still need to update a new "Tables.js" file to include the new vocab import. If that might be a better way, I can commit it to this PR as well.

Reply:
I think this looks better yea. I personally like the dynamic loading idea but @shapiromatron I believe I've submitted code that does that in the past and after feedback went back to doing it explicitly (this may have been in LLR not HAWC now that I think about it). Any preference?

Reply:
sure, this seems fine!  I generally like things more explicit if there are securiy implications, but client side, seems fine.  The only thing I don't like about this pattern is it doesn't really handle the case where the vocab doesn't match, you'd get a silent undefined. maybe something like this?

```js

const tables = {ehv: EhvTable, toxref: ToxrefTable},
  Table = tables[this.props.store.config.vocab];

return (
  ...
  {_.isUndefined(Table)?<p>ERROR - unknown vocabulary</p>:<Table/>}
  ...
)
```